pkg/network/filtermanager.go

```diff
@@ -69,48 +69,43 @@ func (fm *filterManager) InitializeReadFilters() bool {
 	if len(fm.upstreamFilters) == 0 {
 		return false
 	}
-
-	fm.onContinueReading(nil)
-	return true
-}
-
-func (fm *filterManager) onContinueReading(filter *activeReadFilter) {
 	var index int
 	var uf *activeReadFilter
-
-	if filter != nil {
-		index = filter.index + 1
-	}
-
+	var status api.FilterStatus
 	for ; index < len(fm.upstreamFilters); index++ {
 		uf = fm.upstreamFilters[index]
 		uf.index = index
 
 		if !uf.initialized {
 			uf.initialized = true
-
-			status := uf.filter.OnNewConnection()
-
+			status = uf.filter.OnNewConnection()
 			if status == api.Stop {
-				return
+				return false
 			}
 		}
+	}
+	return true
+}
 
-		buf := fm.conn.GetReadBuffer()
+func (fm *filterManager) onContinueReading(previousIndex int, buf buffer.IoBuffer) {
```

Comment:
```suggestion
func (fm *filterManager) onContinueReading(index int, buf buffer.IoBuffer) {
```
maybe we can just use `index` here? It looks more simpiler.